src/combinatorics/quiver/single_obj.lean

```diff
@@ -0,0 +1,112 @@
+/-
+Copyright (c) 2022 Antoine Labelle. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Labelle
+-/
+import combinatorics.quiver.cast
+import combinatorics.quiver.symmetric
+
+/-!
+# Single-object quiver
+
+Single object quiver with a given arrows type.
+
+## Main definitions
+
+Given a type `α`, `single_obj α` is `unit` type, whose single object is called `star α`, with
+`quiver` structure such that `star α ⟶ star α` is the type `α`.
+An element `x : α` can be reinterpreted as an element of `star α ⟶ star α` using
+`to_hom`.
+More generally, a list of elements of `a` can be reinterpreted as a path from `star α` to
+itself using `path_equiv_list`.
+-/
+
+namespace quiver
+
+/-- Type tag on `unit` used to define single-object quivers. -/
+@[nolint unused_arguments]
+def single_obj (α : Type*) : Type := unit
+
+namespace single_obj
+
+variables (α β γ : Type*)
+
+instance : quiver (single_obj α) := ⟨λ _ _, α⟩
+
+/-- The single object in `single_obj α`. -/
+def star : single_obj α := unit.star
+
+instance : inhabited (single_obj α) := ⟨star α⟩
+
+variables {α β γ}
+
+/-- Equip `single_obj α` with a reverse operation. -/
+@[reducible] -- See note [reducible non-instances]
+def has_reverse (rev : α → α) : has_reverse (single_obj α) := ⟨λ _ _, rev⟩
+
+/-- Equip `single_obj α` with an involutive reverse operation. -/
+@[reducible] -- See note [reducible non-instances]
+def has_involutive_reverse (rev : α → α) (h : function.involutive rev) :
+  has_involutive_reverse (single_obj α) :=
+{ to_has_reverse := has_reverse rev,
+  inv' := λ _ _, h}
+
+/-- The type of arrows from `star α` to itself is equivalent to the original type `α`. -/
+@[simps] def to_hom : α ≃ (star α ⟶ star α) := equiv.refl _
+
+/--
+Prefunctors between two `single_obj` quivers correspond to functions between the corresponding
+arrows types.
+-/
+def map_fun :
+  (α → β) ≃ (single_obj α ⥤q single_obj β) :=
+{ to_fun := λ f, ⟨id, λ _ _, f⟩,
+  inv_fun := λ f a, f.map (to_hom a),
+  left_inv := λ _, rfl,
+  right_inv :=  λ f, by cases f; obviously }
+
+lemma map_fun_id : map_fun id = 𝟭q (single_obj α) := rfl
+
+@[simp] lemma map_fun_symm_id :
+  map_fun.symm (𝟭q (single_obj α)) = id := rfl
+
+lemma map_fun_comp (f : α → β) (g : β → γ) :
+  map_fun (g ∘ f) = (map_fun f ⋙q map_fun g) := rfl
```

Comment:
```suggestion
  map_fun (g ∘ f) = map_fun f ⋙q map_fun g := rfl
```

Reply:
I get an error if I remove these parentheses, so I think they are needed.

Reply:
Then you should fix the precedence of the `⋙q` notation. This definitely should work.